packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -49,10 +74,30 @@ export class NetworkListener extends BaseListener implements INetworkListener {
   private readonly loadBalancer: INetworkLoadBalancer;
 
   constructor(scope: cdk.Construct, id: string, props: NetworkListenerProps) {
+    const proto = props.protocol || Protocol.Tcp;
```

Comment:
This still does not have the behavior of defaulting to TLS if `proto` is unspecified and the user specifies certificates.